File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        HashSet<Integer> hashSet = new HashSet<Integer>();
        for(int i=0; i<nums.length; i++){
            hashSet.add(nums[i]);
        }
        return hashSet.size()<nums.length/2 ? hashSet.size() : nums.length/2;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <bits/stdc++.h>
using namespace std;
map<int,int> mpp;
int solution(vector<int> nums)
{
    for(int num : nums){
        mpp[num] += 1;
    }
    
    return mpp.size() < nums.size()/2 ? mpp.size() : nums.size()/2;
}